### PR TITLE
feat: added logger interface

### PR DESCRIPTION
**Pull request by DmitriyLewen**

### CODE REVIEW

File: log/logger.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	Debugf(template string, args ...interface{})
+	Fatal(args ...interface{})
+	Fatalf(template string, args ...interface{})
+}
+
+func initDefaultLogger() LoggerType {
+	return stdoutlogger.NewLogger()
+}
+
+func SetLogger(loggerType LoggerType) {
```
Comment (AndreyLevchenko) [inline log/logger.go:24]: `func SetLogger(logger LoggerType)`

File: outputs/email.go [hunk 2]
```diff
@@ -88,25 +88,25 @@ func (email *EmailOutput) Send(content map[string]string) error {
 	auth := smtp.PlainAuth("", email.User, email.Password, email.Host)
 	err := smtp.SendMail(email.Host+":"+strconv.Itoa(email.Port), auth, email.Sender, recipients, []byte(msg))
 	if err != nil {
-		log.Println("SendMail Error:", err)
-		log.Printf("From: %q, to %v via %q", email.Sender, email.Recipients, email.Host)
+		log.Logger.Error("SendMail Error:", err)
```
Comment (AndreyLevchenko) [inline outputs/email.go:91]: placeholder is missed

File: outputs/email.go [hunk 3]
```diff
@@ -88,25 +88,25 @@ func (email *EmailOutput) Send(content map[string]string) error {
...
+		log.Logger.Errorf("From: %q, to %v via %q", email.Sender, email.Recipients, email.Host)
 		return err
 	}
-	log.Println("Email was sent successfully!")
+	log.Logger.Infof("Email was sent successfully!")
```
Comment (AndreyLevchenko) [inline outputs/email.go:95]: it's debug level

File: log/logger.go [hunk 4]
```diff
@@ -0,0 +1,26 @@
+package log
+
+import "github.com/aquasecurity/postee/log/stdoutlogger"
+
+var Logger LoggerType = initDefaultLogger()
+
+type LoggerType interface {
+	Info(args ...interface{})
```
Comment (AndreyLevchenko) [inline log/logger.go:8]: method to set minimal logging level is missed
like `SetLogLevel("DEBUG")` to write debug messages to log. Or at least flags to enable debug or disable logs at all. Like in trivy

File: log/stdoutlogger/stdoutlogger.go [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+	errorLevel  = colorRed + "   [ERROR]   " + colorReset
+	debugLevel  = colorPurple + "   [DEBUG]   " + colorReset
+	fatalLevel  = colorRed + "   [FATAL]   " + colorReset
+)
+
+type StdOutLogger struct {
+	logger log.Logger
+}
+
+func NewLogger() StdOutLogger {
```
Comment (AndreyLevchenko) [inline log/stdoutlogger/stdoutlogger.go:26]: Can we use zap internally?

File: outputs/email.go [hunk 6]
```diff
@@ -88,25 +88,25 @@ func (email *EmailOutput) Send(content map[string]string) error {
...
 	return nil
 }
 
 func sendViaMxServers(from, subj, msg string, recipients []string) {
 	for _, rcpt := range recipients {
 		at := strings.LastIndex(rcpt, "@")
 		if at < 0 {
-			log.Printf("%q isn't email", rcpt)
+			log.Logger.Errorf("%q isn't email", rcpt)
```
Comment (AndreyLevchenko) [inline outputs/email.go:103]: isn't valid email

File: outputs/email.go [hunk 7]
```diff
@@ -117,11 +117,11 @@ func sendViaMxServers(from, subj, msg string, recipients []string) {
...
 
 			if err := smtp.SendMail(mx.Host+":25", nil, from, []string{rcpt}, []byte(message)); err != nil {
-				log.Printf("SendMail error to %q via %q", rcpt, mx.Host)
-				log.Print(err)
+				log.Logger.Errorf("SendMail error to %q via %q", rcpt, mx.Host)
+				log.Logger.Error(err)
 				continue
 			}
-			log.Printf("The message to %q was sent successful via %q!", rcpt, mx.Host)
+			log.Logger.Infof("The message to %q was sent successful via %q!", rcpt, mx.Host)
```
Comment (AndreyLevchenko) [inline outputs/email.go:124]: it's debug probably

File: outputs/slack.go [hunk 8]
```diff
@@ -78,7 +78,7 @@ func (slack *SlackOutput) Send(input map[string]string) error {
 	rawBlock := make([]data.SlackBlock, 0)
 	err := json.Unmarshal([]byte(body), &rawBlock)
 	if err != nil {
-		log.Printf("Unmarshal slack sending error: %v", err)
+		log.Logger.Errorf("Unmarshal slack sending error: %v", err)
```
Comment (AndreyLevchenko) [inline outputs/slack.go:81]: Unable to parse json

File: regoservice/eval.go [hunk 9]
```diff
@@ -78,7 +78,7 @@ func (regoEvaluator *regoEvaluator) Eval(in map[string]interface{}, serverUrl st
 
 func getFirstElement(context map[string]interface{}, key string) interface{} {
 	for _, v := range context {
-		log.Printf("checking: %s ...\n", key)
+		log.Logger.Infof("checking: %s ...\n", key)
```
Comment (AndreyLevchenko) [inline regoservice/eval.go:81]: make it debug

File: router/router.go [hunk 10]
```diff
@@ -159,32 +159,32 @@ func (ctx *Router) applyTenantCfg(tenant *data.TenantSettings, synchronous bool)
...
 		ctx.quit <- struct{}{}
 	}
 
-	log.Printf("quit notified")
+	log.Logger.Info("quit notified")
 
 	if ctx.ticker != nil {
 		ctx.stopTicker <- struct{}{}
-		log.Printf("stopTicker notified")
+		log.Logger.Info("stopTicker notified")
```
Comment (AndreyLevchenko) [inline router/router.go:187]: debug

File: router/router.go [hunk 11]
```diff
@@ -159,32 +159,32 @@ func (ctx *Router) applyTenantCfg(tenant *data.TenantSettings, synchronous bool)
...
 
-	log.Printf("ctx.quit %v\n", ctx.quit)
+	log.Logger.Infof("ctx.quit %v\n", ctx.quit)
 
 	if ctx.quit != nil {
 		ctx.quit <- struct{}{}
 	}
 
-	log.Printf("quit notified")
+	log.Logger.Info("quit notified")
```
Comment (AndreyLevchenko) [inline router/router.go:183]: debug

File: router/router.go [hunk 12]
```diff
@@ -580,16 +580,16 @@ func (ctx *Router) HandleRoute(routeName string, in []byte) {
 	for _, outputName := range r.Outputs {
 		pl, ok := ctx.outputs[outputName]
 		if !ok {
-			log.Printf("route %q contains an output %q, which doesn't enable now.", routeName, outputName)
+			log.Logger.Errorf("route %q contains an output %q, which doesn't enable now.", routeName, outputName)
```
Comment (AndreyLevchenko) [inline router/router.go:583]: Route %q contains reference to not enabled output %q.

**Review by AndreyLevchenko — CHANGES_REQUESTED** [review]

File: router/router.go (hunk 12 above)
Comment (DmitriyLewen) [reply router/router.go]: Done

File: router/router.go (hunk 11 above)
Comment (DmitriyLewen) [reply router/router.go]: Done

File: router/router.go (hunk 10 above)
Comment (DmitriyLewen) [reply router/router.go]: Done

File: regoservice/eval.go (hunk 9 above)
Comment (DmitriyLewen) [reply regoservice/eval.go]: Done

File: outputs/slack.go (hunk 8 above)
Comment (DmitriyLewen) [reply outputs/slack.go]: Done

File: outputs/email.go (hunk 7 above)
Comment (DmitriyLewen) [reply outputs/email.go]: Done

File: outputs/email.go (hunk 6 above)
Comment (DmitriyLewen) [reply outputs/email.go]: Done

File: outputs/email.go (hunk 3 above)
Comment (DmitriyLewen) [reply outputs/email.go]: Done

File: outputs/email.go (hunk 2 above)
Comment (DmitriyLewen) [reply outputs/email.go]: Done

File: dbservice/boltdb/checker.go [hunk 13]
```diff
@@ -38,7 +38,7 @@ func (boltDb *BoltDb) CheckSizeLimit() {
 		}
 		return nil
 	}); err != nil {
-		log.Println("Error a check of db size:", err)
+		log.Logger.Errorf("Error a check of db size: %v", err)
```
[the remark this answers is not in the document]
Comment (DmitriyLewen) [reply dbservice/boltdb/checker.go]: Done

File: log/stdoutlogger/stdoutlogger.go (hunk 5 above)
Comment (DmitriyLewen) [reply log/stdoutlogger/stdoutlogger.go]: Done

File: log/logger.go (hunk 4 above)
Comment (DmitriyLewen) [reply log/logger.go:13]: Done

File: log/logger.go (hunk 1 above)
Comment (DmitriyLewen) [reply log/logger.go]: Done

**Review by AndreyLevchenko — APPROVED** [review]

**Pull request merged**